### PR TITLE
AML SDK fixture Reamd update

### LINKED ISSUE #40: Azure ML SDK mocking and unit testing
Some of this work is demonstrated in [this sample](https://github.com/liupeirong/MLOpsManufacturing/tree/main/samples/non-python-preprocess/ml_service/tests/pipelines). This item is to 
1. Make it more generic and reusable. Put it in the `common` folder since it's a cross-cutting concern.
2. Create more unit tests in samples to demonstrate its usage.

### PR DESCRIPTION
**Pull request by angie4u**
This readme explains how to use AML SDK fixture with your unit test code.
I still need a lot of improvement(such as - how to use spy and so on), but I created this PR because I wanted to get a review before Dragon Crew left.
Any comments are welcome! Please check if overall flow of this readme make sense, whether it is easy to understand.

### CODE REVIEW

File: common/pytest-fixtures/README.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+                                          subscription_id=subscription_id)
+
+    # Check if Pipeline publish was called with arguments
+    mock_pipeline_publish.assert_called_once_with(name=preprocessing_pipeline_name,
+                                                  description="Data preprocessing"
+                                                              " OS cmd pipeline",
+                                                  version=build_id)
+    ```
+
+    This code shows how to make an assertion using an imported mock objects.
```
Comment (h2floh) [inline common/pytest-fixtures/README.md]: You could add here a comment where the workspace_name, rg and sub_id are set/hard coded in the fixture file.

File: common/pytest-fixtures/README.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+### Fixture for mocking AML SDK with Env util
+
+> [This fixture](./test_aml_mock_fixtures_env.py) is **dependent on the ml_service.util.env_variables.Env class**
+used in [non-python-preprocess sample](/samples/non-python-preprocess/ml_service/util/env_variables.py).
+
+This fixture is for the case of referencing environment variables using env util.
+
+### How to use
+
+TODO
```
Comment (h2floh) [inline common/pytest-fixtures/README.md]: You can basically copy the advise from the independent fixture and change the comment that workspace_name, rg, sub_id are now changed at a different place (within mock for EnvUtil)

**Review by h2floh — COMMENTED** [review]
Looks great @angie4u ! Please see the other comments. Please also have a look and try to fix the linter errors [here](https://github.com/liupeirong/MLOpsManufacturing/pull/57/checks?check_run_id=1915102420). Basically removing unused imports.

File: samples/non-python-preprocess/devops_pipelines/dependency-update-template.yml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Pipeline template to update the custom docker image with latest dependencies
+steps:
+- script: |   
+   conda env update -f ${{ parameters.workingDirectory }}/local_development/dev_dependencies.yml
```
Comment (h2floh) [inline samples/non-python-preprocess/devops_pipelines/dependency-update-template.yml]:
```suggestion
   conda env update -f local_development/dev_dependencies.yml -n ci
```

File: samples/non-python-preprocess/devops_pipelines/dependency-update-template.yml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# Pipeline template to update the custom docker image with latest dependencies
+steps:
+- script: |   
+   conda env update -f local_development/dev_dependencies.yml -n ci
+  workingDirectory: ${{ parameters.workingDirectory }}
+  displayName: 'Download pip packages with latest dependency file'
+- script: |
```
Comment (angie4u) [inline samples/non-python-preprocess/devops_pipelines/dependency-update-template.yml:7]: @h2floh Do you think can I leave this code to check required package installed or is it better to remove?

Comment (h2floh) [reply samples/non-python-preprocess/devops_pipelines/dependency-update-template.yml:7]: Like it, will help debugging if the pipeline fails!

**Review by h2floh — APPROVED** [review]

**Pull request merged**